phoenix-core/src/main/java/org/apache/phoenix/hbase/index/write/TrackingParallelWriterIndexCommitter.java

```diff
@@ -242,8 +243,13 @@ private void throwFailureIfDone() throws SingleIndexWriteFailureException {
         // if any of the tasks failed, then we need to propagate the failure
         if (failures.size() > 0) {
             // make the list unmodifiable to avoid any more synchronization concerns
-            throw new MultiIndexWriteFailureException(Collections.unmodifiableList(failures),
+            MultiIndexWriteFailureException exception = new MultiIndexWriteFailureException(Collections.unmodifiableList(failures),
                     disableIndexOnFailure && PhoenixIndexFailurePolicy.getDisableIndexOnFailure(env));
+            if (disableIndexOnFailure)
+                throw exception;
+            else {
+                throw new DoNotRetryIOException(exception.getMessage());
```

Comment:
Shouldn't we wrap the MIWFException in the DNRIE so that higher up can see the stack trace / any inner exceptions rather than just copying the message?

Reply:
I will do that.

Reply:
Done